Fix pubkey auth for users whose home dir is on a NFS share with root squash.

--- a/svr-authpubkey.c
+++ b/svr-authpubkey.c
@@ -415,9 +415,21 @@
 	char* filename = NULL; 
 	int ret = DROPBEAR_FAILURE;
 	unsigned int len;
+	uid_t origuid;
+	gid_t origgid;
 
 	TRACE(("enter checkpubkeyperms"))
 
+#if DROPBEAR_SVR_MULTIUSER
+	/* check file permissions as the authenticating user. */
+	origuid = getuid();
+	origgid = getgid();
+	if ((setegid(ses.authstate.pw_gid)) < 0 ||
+		(seteuid(ses.authstate.pw_uid)) < 0) {
+		dropbear_exit("Failed to set euid");
+	}
+#endif
+
 	if (ses.authstate.pw_dir == NULL) {
 		goto out;
 	}
@@ -455,6 +467,13 @@
 out:
 	m_free(filename);
 
+#if DROPBEAR_SVR_MULTIUSER
+	if ((seteuid(origuid)) < 0 ||
+		(setegid(origgid)) < 0) {
+		dropbear_exit("Failed to revert euid");
+	}
+#endif
+
 	TRACE(("leave checkpubkeyperms"))
 	return ret;
 }
